tools/unsupervised_dataset/Crawler/Crawler.py

```diff
@@ -0,0 +1,101 @@
+import requests
+import queue
+import CQNode
+import Extractor
+import CrawlerLog
+
+class Crawler:
+    """ Contains the functions and logic to run and coordinate the crawling process. Given
+        initial starting URLs and maximum size, the crawler will explore websites to look
+        for SQL queries.
+    """
+    
+    def __init__(self, links, maxDepth=3, maxSize=500):
+        """ Initializes the crawler and instance variables.
+        
+        Args:
+            links: The root URLs to begin crawling from. Can be one or more.
+            maxDepth: The maximum depth for the crawler to explore.
+            maxSize: THe maximum number of links the crawler should explore.
+    
+        """
+        self.linkQueue = queue.Queue()
+        self.seen = set([])
+        self.maxDepth = maxDepth
+        self.maxSize = maxSize
+        self.log = CrawlerLog.CrawlerLog()
+        self.count = 0
+        
+        for link in links:
+            self.linkQueue.put(CQNode.CQNode(link, 0))
+            self.seen.add(link)
+        
+    def crawl(self):
+        """ Begins the crawling process using variables set earlier. Extracts
+            queries by locating website-specific HTML tags or searching for
+            common expression patterns. Writes queries to output after finishing
+            each site.
+        """
+        
+        while (self.linkQueue.empty() == False):
+            # Retrieve the next link in the queue
+            nextNode = self.linkQueue.get()
+            nodeURL = nextNode.getURL()
+            nodeDepth = nextNode.getDepth()
+            
+            # Check if crawler has exceeded maximum depth or maximum count
+            if nodeDepth >= self.maxDepth or self.count >= self.maxSize:
+                print("Crawled {0} websites.".format(self.count))
```

Comment:
Instead of printing to console, try to print to some logger, such as https://docs.python.org/2/library/logging.html